src/mono/mono/mini/mini-runtime.c

```diff
@@ -3158,9 +3158,8 @@ mono_llvmonly_runtime_invoke (MonoMethod *method, RuntimeInvokeInfo *info, void
 {
 	MonoMethodSignature *sig = info->sig;
 	MonoObject *(*runtime_invoke) (MonoObject *this_obj, void **params, MonoObject **exc, void* compiled_method);
-	gboolean retval_malloc = FALSE;
-	gpointer retval_ptr;
-	guint8 retval [MONO_SIZEOF_DYN_CALL_RET_BUF];
+	int32_t retval_size = MONO_SIZEOF_DYN_CALL_RET_BUF;
```

Comment:
Could the initial size be `SIZEOF_VOID_P` in order to avoid unnecessary stack allocation ?